src/pages/wallet-overview/wallet-overview.ts

```diff
@@ -16,89 +19,187 @@ import { ICurrency } from '../../models/ICurrency';
 @Component({
   selector: "page-wallet-overview",
   templateUrl: "wallet-overview.html",
+  animations: [
+    trigger('visibilityChanged', [
+      state('shown', style({ opacity: 1 })),
+      state('hidden', style({ opacity: 0 , display: "none"})),
+      transition('* => *', animate('500ms'))
+    ])
+  ]
 })
 export class WalletOverviewPage {
   @ViewChild('doughnutCanvas') doughnutCanvas;
-  title = "Wallet Overview";
-  pickedCurrency = "$";
-  yellows: number;
-  blues: number;
-  currencyAmount: number;
-  smiloWorthPerCoin: number;
-  smiloPayWorthPerCoin: number;
+  pickedCurrency: string;
   doughnutChart: any;
   wallets: any = [];
   currenciesForDoughnutCanvas: any = [];
   currenciesForDoughnutCanvasCurrencies: any = [];
   currentWallet: any;
   currentWalletIndex = 0;
-  legendList: string[];
+  legendList: string[] = [];
   availableCurrencies: string[] = [];
+  showFundsStatus: boolean = true;
+  twoFactorStatus: boolean = false;
+  walletFundsVisibility: string = 'shown';
+  loading: Loading;
+  loadingError: Loading;
+
 
   constructor(public navCtrl: NavController, 
               public navParams: NavParams, 
-              public walletService: WalletService) {
-    this.currentWallet = "ETm9QUJLVdJkTqRojTNqswmeAQGaofojJJ";
+              public walletService: WalletService,
+              public alertCtrl: AlertController,
+              public toastCtrl: ToastController,
+              public loadingCtrl: LoadingController) {
     this.getAllWallets();
     this.getAvailableCurrencies();
   }
 
   ionViewDidLoad() {
-    console.log("ionViewDidLoad WalletOverviewPage");
+    
+  }
+
+  fundsSwitch() {
+    if (this.walletFundsVisibility === "shown") {
+      this.walletFundsVisibility = "hidden";
+      this.showFundsStatus = false;
+    } else if (this.walletFundsVisibility === "hidden") {
+      this.walletFundsVisibility = "shown";
+      this.showFundsStatus = true;
+    }
+  }
+
+  twoFactorStatusSwitch() {
+    if (this.twoFactorStatus) {
+      this.twoFactorStatus = false;
+    } else {
+      this.twoFactorStatus = true;
+    }
+  }
+
+  backupWalletClick() {
+
+  }
+
+  deleteWalletClick() {
+    if (this.currentWallet === null || this.currentWallet === undefined) {
+      return null;
+    }
+    const confirm = this.alertCtrl.create({
+      title: 'Delete wallet',
+      message: "Are you <b>sure</b> you want to delete this wallet ('" + this.currentWallet.publicKey + "')?",
+      buttons: [
+        {
+          text: 'No, cancel',
+          handler: () => {
+            
+          }
+        },
+        {
+          text: 'Yes, delete',
+          cssClass: 'deleteButtonCss',
+          handler: () => {
+            this.deleteSelectedWallet(this.currentWallet.publicKey);
+          }
+        }
+      ]
+    });
+    if (confirm !== null) {
+      confirm.present();
+    } else {
+      return null;
+    }
+  }
+
+  deleteSelectedWallet(publicKey: string) {
```

Comment:
Right now we delete a wallet by passing the public key of the wallet. Why was this chosen instead of simply passing the wallet object as parameter?